<!doctype html><html lang=en dir=ltr><head><script async src="https://www.googletagmanager.com/gtag/js?id=G-9TMPT34HZ4"></script><script>window.dataLayer=window.dataLayer||[];function gtag(){dataLayer.push(arguments);}
gtag('js',new Date());gtag('config','G-9TMPT34HZ4');</script><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge,chrome=1"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=generator content="Hugo 0.77.0"><title>Where can I publish a paper describing my Bioconductor package? | Fabrício's personal website</title><meta property="twitter:site" content="@almeidasilvaf"><meta property="twitter:creator" content="@almeidasilvaf"><meta name=description content="Check out where Bioc developers have published their papers"><meta property="og:site_name" content="Fabrício's personal website"><meta property="og:title" content="Where can I publish a paper describing my Bioconductor package? | Fabrício's personal website"><meta property="og:description" content="Check out where Bioc developers have published their papers"><meta property="og:type" content="page"><meta property="og:url" content="/blog/2022-01-03-bioc_publications/"><meta property="og:locale" content="en"><meta property="og:image" content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta property="twitter:card" content="summary_large_image"><meta name=twitter:image content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta itemprop=name content="Where can I publish a paper describing my Bioconductor package?"><meta itemprop=description content="Motivation     When I developed BioNERO, my first R/Bioconductor package, I didn’t know to which journals I could submit the paper describing it. Since then, I’ve seen many other R developers that have faced the same issue. To help solve this problem, here I will guide you on how to do some web scraping to find out the main journals where Bioc developers publish papers describing their packages."><meta itemprop=datePublished content="2022-01-02T00:00:00+00:00"><meta itemprop=dateModified content="2022-01-02T00:00:00+00:00"><meta itemprop=wordCount content="1578"><meta itemprop=image content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta itemprop=keywords content><script type=application/javascript>var dnt=(navigator.doNotTrack||window.doNotTrack||navigator.msDoNotTrack);var doNotTrack=(dnt=="1"||dnt=="yes");if(!doNotTrack){window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;ga('create','G-9TMPT34HZ4','auto');ga('set','anonymizeIp',true);ga('send','pageview');}</script><script async src=https://www.google-analytics.com/analytics.js></script><!--[if IE]><script src=//html5shiv.googlecode.com/svn/trunk/html5.js></script><![endif]--><link rel="shortcut icon" href=/img/logo_tree_graph.png type=image/x-icon><link rel=icon href=/img/logo_tree_graph.png type=image/x-icon><link rel=stylesheet href=/style.main.min.306f61a3dc4566d041ed43ec1724f89fc7dfd8c53cd1b942e7b37a3b790db35b.css integrity="sha256-MG9ho9xFZtBB7UPsFyT4n8ff2MU80blC57N6O3kNs1s=" media=screen><script src=/panelset.min.41f62c42d5f25dd80e81136866f4f1baa1bc17669edfd002e75c0e107953789b.js type=text/javascript></script></head>...</head><body><div class="grid-container single"><header class="site-header pt4 pb2 mb4 bb b--transparent ph5 headroom z-max" role=banner><script>MathJax={tex:{inlineMath:[['$','$'],['\\(','\\)']],displayMath:[['$$','$$'],['\\[','\\]']],processEscapes:true,processEnvironments:true},options:{skipHtmlTags:['script','noscript','style','textarea','pre']}};window.addEventListener('load',(event)=>{document.querySelectorAll("mjx-container").forEach(function(x){x.parentElement.classList+='has-jax'})});</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script type=text/javascript id=MathJax-script async src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script><nav class="site-nav db dt-l w-100" role=navigation><a class="site-brand db dtc-l v-mid link no-underline w-100 w-33-l tc tl-l" href=/ title=Home><img src=/img/logo_tree_graph.png class="dib db-l h2 w-auto" alt="Fabrício's personal website"></a><div class="site-links db dtc-l v-mid w-100 w-47-l tc tr-l mt3 mt0-l ttu tracked"><a class="link f6 f5-l dib pv1 ph2" href=/about/ title="About me">About</a>
<a class="link f6 f5-l dib pv1 ph2" href=/project/ title=Softwares>Softwares & Data</a>
<a class="link f6 f5-l dib pv1 ph2" href=/publications/ title=Publications>Publications</a>
<a class="link f6 f5-l dib pv1 ph2" href=/talk/ title=Talks>Talks</a>
<a class="link f6 f5-l dib pv1 ph2" href=/teaching/ title=Teaching>Teaching</a>
<a class="link f6 f5-l dib pv1 ph2" href=/cv/ title=CV>CV</a>
<a class="link f6 f5-l dib pv1 ph2 active" href=/blog/ title=Blog>Blog</a></div></nav></header><main class="page-main pa4" role=main><section class="page-content mw7 center"><article class="post-content pa0 ph4-l"><header class=post-header><h1 class="f1 lh-solid measure-narrow mb3 fw4">Where can I publish a paper describing my Bioconductor package?</h1><p class="f6 measure lh-copy mv1">By Fabrício Almeida-Silva in <a href=/categories/bioinformatics>bioinformatics</a> <a href=/categories/reproducible-research>reproducible research</a> <a href=/categories/bioconductor>bioconductor</a> <a href=/categories/scientific-writing>scientific writing</a> <a href=/categories/rstats>rstats</a></p><p class="f7 db mv0 ttu">January 2, 2022</p></header><section class="post-body pt5 pb4"><script src=/blog/2022-01-03-bioc_publications/index_files/htmlwidgets/htmlwidgets.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/datatables-css/datatables-crosstalk.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/datatables-binding/datatables.js></script><script src=/blog/2022-01-03-bioc_publications/index_files/jquery/jquery-3.6.0.min.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/dt-core/css/jquery.dataTables.min.css rel=stylesheet><link href=/blog/2022-01-03-bioc_publications/index_files/dt-core/css/jquery.dataTables.extra.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/dt-core/js/jquery.dataTables.min.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/crosstalk/css/crosstalk.min.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/crosstalk/js/crosstalk.min.js></script><h2 id=motivation>Motivation
<a href=#motivation><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>When I developed
<a href=https://bioconductor.org/packages/release/bioc/html/BioNERO.html target=_blank rel=noopener>BioNERO</a>, my first R/Bioconductor package, I didn’t know to which journals I could submit the paper describing it. Since then, I’ve seen many other R developers that have faced the same issue. To help solve this problem, here I will guide you on how to do some web scraping to find out the main journals where Bioc developers publish papers describing their packages.</p><h2 id=extracting-citation-information-from-packages-landing-pages>Extracting citation information from packages’ landing pages
<a href=#extracting-citation-information-from-packages-landing-pages><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>All Bioconductor packages have a landing page where there is a citation field. This field is created from a CITATION file inside the package directory or automatically from information in the DESCRIPTION file. First, let’s write two helper functions:</p><ul><li><code>get_bioc_citation</code>, a function that scrapes packages’ landing pages to get citation information.</li><li><code>create_citation_table</code>, a function that creates a tidy data frame containing package name, journal where the paper is published (if that is the case), and year of publication.</li><li><code>process_titles</code>, a function to process journal titles and remove inconsistencies across packages.</li></ul><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#998;font-style:italic>#&#39; Extract citation from Bioconductor landing page</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @param package Character of package name.</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @return Character of citation text.</span>
<span style=color:#998;font-style:italic>#&#39; @noRd</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom rvest html_elements html_text2</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom httr GET content</span>
get_bioc_citation <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>function</span>(package <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>NULL</span>) {
    url <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>paste0</span>(<span style=color:#d14>&#34;https://bioconductor.org/packages/release/bioc/citations/&#34;</span>,
                  package, <span style=color:#d14>&#34;/citation.html&#34;</span>)
    request <span style=color:#000;font-weight:700>&lt;-</span> httr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>GET</span>(url)
    parsed_page <span style=color:#000;font-weight:700>&lt;-</span> httr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>content</span>(request, as <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;parsed&#34;</span>, encoding <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;UTF-8&#34;</span>)
    <span style=color:#900;font-weight:700>if</span>(<span style=color:#d14>&#34;xml_node&#34;</span> <span style=color:#000;font-weight:700>%in%</span> <span style=color:#900;font-weight:700>class</span>(parsed_page)) {
        citations <span style=color:#000;font-weight:700>&lt;-</span> rvest<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>html_elements</span>(parsed_page, <span style=color:#d14>&#34;p&#34;</span>)
        citations <span style=color:#000;font-weight:700>&lt;-</span> rvest<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>html_text2</span>(citations)
    } else {
        citations <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#000;font-weight:700>NA</span>
    }
    <span style=color:#900;font-weight:700>return</span>(citations)
}

<span style=color:#998;font-style:italic>#&#39; Create a table of citations for each Bioc package</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @param packages Character vector of package names.</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @return A data frame with the following variables:</span>
<span style=color:#998;font-style:italic>#&#39; \describe{</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Package}{Chaarcter of package name.}</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Journal}{Character of journal name.}</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Year}{Numeric of journal year.}</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Citation}{Character of full citation.}</span>
<span style=color:#998;font-style:italic>#&#39; }</span>
<span style=color:#998;font-style:italic>#&#39; @noRd</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom stringr str_detect str_extract</span>
create_citation_table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>function</span>(packages <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>NULL</span>) {
    table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>Reduce</span>(rbind, <span style=color:#900;font-weight:700>lapply</span>(packages, <span style=color:#900;font-weight:700>function</span>(x) {
        df <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>data.frame</span>(Citation <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>get_bioc_citation</span>(x))
        df<span style=color:#000;font-weight:700>$</span>Package <span style=color:#000;font-weight:700>&lt;-</span> x
        <span style=color:#900;font-weight:700>return</span>(df)
    }))
    
    year <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;([0-9]{4})&#34;</span>
    table<span style=color:#000;font-weight:700>$</span>Year <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>as.numeric</span>(stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_extract</span>(table<span style=color:#000;font-weight:700>$</span>Citation, year))
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#39;.*” &#39;</span>, <span style=color:#d14>&#39;&#39;</span>, table<span style=color:#000;font-weight:700>$</span>Citation)
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#34;\\*{3} &#34;</span>, <span style=color:#d14>&#34;&#34;</span>, journal)
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#34;(,|\\.).*&#34;</span>, <span style=color:#d14>&#34;&#34;</span>, journal)
    table<span style=color:#000;font-weight:700>$</span>Journal <span style=color:#000;font-weight:700>&lt;-</span> journal
    idx <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>which</span>(stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_detect</span>(table<span style=color:#000;font-weight:700>$</span>Citation, <span style=color:#d14>&#34;github|R package version&#34;</span>))
    table<span style=color:#000;font-weight:700>$</span>Journal[idx] <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#000;font-weight:700>NA</span>
    table <span style=color:#000;font-weight:700>&lt;-</span> table[, <span style=color:#900;font-weight:700>c</span>(<span style=color:#d14>&#34;Package&#34;</span>, <span style=color:#d14>&#34;Journal&#34;</span>, <span style=color:#d14>&#34;Year&#34;</span>, <span style=color:#d14>&#34;Citation&#34;</span>)]
    table <span style=color:#000;font-weight:700>&lt;-</span> table[<span style=color:#000;font-weight:700>!</span><span style=color:#900;font-weight:700>is.na</span>(table<span style=color:#000;font-weight:700>$</span>Journal), ]
    <span style=color:#900;font-weight:700>return</span>(table)
}

<span style=color:#998;font-style:italic>#&#39; Process journal titles</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @param citation_table Data frame created with \code{create_citation_table}.</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @return The same data frame as in \strong{citation_table}, but with</span>
<span style=color:#998;font-style:italic>#&#39; processed titles in the variable &#39;Journal&#39;.</span>
<span style=color:#998;font-style:italic>#&#39; @noRd</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom dplyr mutate filter</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom stringr str_to_title str_replace_all str_detect str_trim</span>
process_titles <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>function</span>(citation_table <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>NULL</span>) {
    pcit <span style=color:#000;font-weight:700>&lt;-</span> dplyr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>mutate</span>(citation_table, 
                          Journal <span style=color:#000;font-weight:700>=</span> stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_to_title</span>(Journal))
    
    sagmb <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;Statistical Applications In Genetics And Molecular Biology&#34;</span>
    pnas <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;Proceedings Of The National Academy Of Sciences Of The United States Of America&#34;</span>
    ijms <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;International Journal Of Molecular Sciences&#34;</span>
    cmpbiomed <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;Computer Methods And Programs In Biomedicine&#34;</span>
    jcompbio <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;Journal Of Computational Biology&#34;</span>
    pcit <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>mutate</span>(pcit, Journal <span style=color:#000;font-weight:700>=</span> stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_replace_all</span>(
        Journal, <span style=color:#900;font-weight:700>c</span>(<span style=color:#d14>&#34;^Nat$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature&#34;</span>,
                   <span style=color:#d14>&#34;Res$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Research&#34;</span>,
                   <span style=color:#d14>&#34; \\(Oxford.*&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>,
                   <span style=color:#d14>&#34;Bmc&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;BMC&#34;</span>,
                   <span style=color:#d14>&#34;Plos&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;PLoS&#34;</span>,
                   <span style=color:#d14>&#34;Rna&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;RNA&#34;</span>,
                   <span style=color:#d14>&#34;^Nucl$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nucleic Acids Research&#34;</span>,
                   <span style=color:#d14>&#34;^Genome Bio$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Genome Biology&#34;</span>,
                   <span style=color:#d14>&#34;^Genome Biol$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Genome Biology&#34;</span>,
                   <span style=color:#d14>&#34;^F1000$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;F1000Research&#34;</span>,
                   <span style=color:#d14>&#34;F1000research&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;F1000Research&#34;</span>,
                   <span style=color:#d14>&#34;F1000res&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;F1000Research&#34;</span>,
                   <span style=color:#d14>&#34;F1000 Research&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;F1000Research&#34;</span>,
                   <span style=color:#d14>&#34;Stat Appl Genet Mol Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> sagmb,
                   <span style=color:#d14>&#34;Sagmb&#34;</span> <span style=color:#000;font-weight:700>=</span> sagmb,
                   <span style=color:#d14>&#34;Proceedings Of The National Academy of Sciences&#34;</span> <span style=color:#000;font-weight:700>=</span> pnas,
                   <span style=color:#d14>&#34;Proc Natl Acad Sci Usa&#34;</span> <span style=color:#000;font-weight:700>=</span> pnas,
                   <span style=color:#d14>&#34;Pnas&#34;</span> <span style=color:#000;font-weight:700>=</span> pnas,
                   <span style=color:#d14>&#34;Comput Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Computational Biology&#34;</span>,
                   <span style=color:#d14>&#34;Computat Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Computational Biology&#34;</span>,
                   <span style=color:#d14>&#34;Physiol Genomics&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Physiological Genomics&#34;</span>,
                   <span style=color:#d14>&#34;Oxford &#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>,
                   <span style=color:#d14>&#34;Bioinformatics Journal&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Bioinformatics&#34;</span>,
                   <span style=color:#d14>&#34; \\(.*\\)&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>,
                   <span style=color:#d14>&#34;Nat Meth&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature Methods&#34;</span>,
                   <span style=color:#d14>&#34;Nat Commun&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature Communications&#34;</span>,
                   <span style=color:#d14>&#34;Nat Biotech&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature Biotechnology&#34;</span>,
                   <span style=color:#d14>&#34;Nat Biotechnol&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature Biotechnology&#34;</span>,
                   <span style=color:#d14>&#34;Nar Genomics And Bioinformatics&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nucleic Acids Research&#34;</span>,
                   <span style=color:#d14>&#34;Mol Cell Proteomics&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Molecular &amp; Cellular Proteomics&#34;</span>,
                   <span style=color:#d14>&#34;Methods Inf Med&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Methods Of Information In Medicine&#34;</span>,
                   <span style=color:#d14>&#34;Jco.*&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Journal Of Clinical Oncology&#34;</span>,
                   <span style=color:#d14>&#34;Jasa&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Journal Of the American Statistical Association&#34;</span>,
                   <span style=color:#d14>&#34;J Proteomics&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Journal Of Proteomics&#34;</span>,
                   <span style=color:#d14>&#34;J Comput Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> jcompbio,
                   <span style=color:#d14>&#34;J Computational Biology&#34;</span> <span style=color:#000;font-weight:700>=</span> jcompbio,
                   <span style=color:#d14>&#34;J Mach Learn&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Journal Of Machine Learning&#34;</span>,
                   <span style=color:#d14>&#34;Int J Mol Sci&#34;</span> <span style=color:#000;font-weight:700>=</span> ijms,
                   <span style=color:#d14>&#34;Comput Methods Programs Biomed&#34;</span> <span style=color:#000;font-weight:700>=</span> cmpbiomed,
                   <span style=color:#d14>&#34;Syst Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Systems Biology&#34;</span>,
                   <span style=color:#d14>&#34;Mol Systems Biology&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Molecular Systems Biology&#34;</span>,
                   <span style=color:#d14>&#34;Algorithms Mol Biol&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Algorithms For Molecular Biology&#34;</span>
                   ))
    )
    pcit <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>mutate</span>(pcit, Journal <span style=color:#000;font-weight:700>=</span> stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_trim</span>(Journal))
    remove <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;Biorxiv|Arxiv|Doi|Preprint|Submitted|In Preparation|Error|Http&#34;</span>
    pcit <span style=color:#000;font-weight:700>&lt;-</span> dplyr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>filter</span>(pcit, 
                          <span style=color:#000;font-weight:700>!</span>stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_detect</span>(Journal, remove))
    
    <span style=color:#900;font-weight:700>return</span>(pcit)
}
</code></pre></div><p>Now, we can create the citation table.</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#900;font-weight:700>suppressPackageStartupMessages</span>(<span style=color:#900;font-weight:700>library</span>(tidyverse))

<span style=color:#998;font-style:italic># List available Bioc packages</span>
bioc_repo <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;https://www.bioconductor.org/packages/release/bioc&#34;</span>
bioc_pkgs <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>available.packages</span>(repos <span style=color:#000;font-weight:700>=</span> bioc_repo)[, <span style=color:#d14>&#34;Package&#34;</span>]
</code></pre></div><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#998;font-style:italic># Scrape CITATION field from Bioc landing pages</span>
citation_table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>create_citation_table</span>(bioc_pkgs)

<span style=color:#998;font-style:italic># Process journal names</span>
citation_table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>process_titles</span>(citation_table)
</code></pre></div><h2 id=summary-stats>Summary stats
<a href=#summary-stats><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>Now, let’s count the frequency of packages in each journal and show the top 20 journals based on highest article count.</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>citation_stats <span style=color:#000;font-weight:700>&lt;-</span> citation_table <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>count</span>(Journal) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>arrange</span>(<span style=color:#000;font-weight:700>-</span>n) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>slice_head</span>(n <span style=color:#000;font-weight:700>=</span> <span style=color:#099>20</span>)

citation_stats
</code></pre></div><pre><code>##                                                       Journal   n
## 1                                              Bioinformatics 272
## 2                                          BMC Bioinformatics  90
## 3                                      Nucleic Acids Research  67
## 4                                              Genome Biology  55
## 5                                               F1000Research  31
## 6                                              Nature Methods  30
## 7                                                BMC Genomics  23
## 8                                                    PLoS One  20
## 9                                  PLoS Computational Biology  19
## 10                                            Genome Research  13
## 11                                              Biostatistics  11
## 12                                      Nature Communications  10
## 13                                       Analytical Chemistry   9
## 14                                  Molecular Systems Biology   8
## 15                                                     Nature   8
## 16                                       Nature Biotechnology   8
## 17                            Molecular &amp; Cellular Proteomics   7
## 18 Statistical Applications In Genetics And Molecular Biology   7
## 19                                      Frontiers In Genetics   6
## 20                               Journal Of Proteome Research   6
</code></pre><p>Exploring it visually:</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>bioc_logo <span style=color:#000;font-weight:700>&lt;-</span> png<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>readPNG</span>(
    here<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>here</span>(<span style=color:#d14>&#34;content&#34;</span>, <span style=color:#d14>&#34;blog&#34;</span>, 
               <span style=color:#d14>&#34;2022-01-03-bioc_publications&#34;</span>, 
               <span style=color:#d14>&#34;featured-bioc.png&#34;</span>), 
    native <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>TRUE</span>)

last_updated <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>format</span>(<span style=color:#900;font-weight:700>Sys.Date</span>(), <span style=color:#d14>&#34;%Y-%m-%d&#34;</span>)
xmax <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>max</span>(citation_stats<span style=color:#000;font-weight:700>$</span>n) <span style=color:#000;font-weight:700>+</span> <span style=color:#099>30</span>
xmax <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>round</span>(xmax <span style=color:#000;font-weight:700>/</span> <span style=color:#099>10</span>) <span style=color:#000;font-weight:700>*</span> <span style=color:#099>10</span>

<span style=color:#900;font-weight:700>ggplot</span>(citation_stats, <span style=color:#900;font-weight:700>aes</span>(x <span style=color:#000;font-weight:700>=</span> n, y <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>reorder</span>(Journal, n))) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>geom_col</span>() <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>geom_text</span>(<span style=color:#900;font-weight:700>aes</span>(label <span style=color:#000;font-weight:700>=</span> n), hjust <span style=color:#000;font-weight:700>=</span> <span style=color:#099>-0.3</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>xlim</span>(<span style=color:#099>0</span>, <span style=color:#099>300</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>labs</span>(title <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Where are papers associated with BioC packages published?&#34;</span>,
         subtitle <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>paste0</span>(<span style=color:#d14>&#34;Last updated: &#34;</span>, last_updated),
         x <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Number of papers&#34;</span>, y <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>theme_bw</span>() <span style=color:#000;font-weight:700>+</span>
    patchwork<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>inset_element</span>(bioc_logo,
                             left <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.5</span>,
                             top <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.55</span>,
                             right <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.95</span>,
                             bottom <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.3</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>theme_void</span>()
</code></pre></div><p><img src=/blog/2022-01-03-bioc_publications/index_files/figure-html/unnamed-chunk-3-1.png width=864></p><p>And voilà! In case you want to explore the whole table, here it is:</p><div id=htmlwidget-1 style=width:100%;height:auto class="datatables html-widget"></div><script type=application/json data-for=htmlwidget-1>{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200"],["Bioinformatics","BMC Bioinformatics","Nucleic Acids Research","Genome Biology","F1000Research","Nature Methods","BMC Genomics","PLoS One","PLoS Computational Biology","Genome Research","Biostatistics","Nature Communications","Analytical Chemistry","Molecular Systems Biology","Nature","Nature Biotechnology","Molecular &amp; Cellular Proteomics","Statistical Applications In Genetics And Molecular Biology","Frontiers In Genetics","Journal Of Proteome Research","Nature Genetics","BMC Systems Biology","Journal Of Statistical Software","Scientific Reports","Briefings In Bioinformatics","Proceedings Of The National Academy Of Sciences","Proceedings Of The National Academy Of Sciences Of The United States Of America","Annals Of Applied Statistics","Biometrics","Cell Reports","Cytometry Part A","Epigenetics &amp; Chromatin","Genomics","Johns Hopkins University","Molecular Biology And Evolution","Peerj","Aibar S","Algorithms For Molecular Biology","Analytica Chimica Acta","Animal Genetics","Biomed Research International","Clinical Cancer Research","Computer Methods And Programs In Biomedicine","Epigenomics","Gigascience","Human Mutation","International Journal Of Molecular Sciences","J","Journal Of Biomedical Informatics","Journal Of Clinical Oncology","Journal Of Machine Learning Research","Luo X","Metabolites","Methods","Methods In Ecology And Evolution","Methods In Molecular Biology","Microbiome","Mol","Nature Protocols","Omics: A Journal Of Integrative Biology","Shokoohi F","Source Code For Biology And Medicine","The Journal Of Open Source Software","","10","Aging","Algorithmica","American Journal Of Human Genetics","Ann","Applications In Genetics And Molecular Biology","Applications In Plant Sciences","Baek M","Bbagrm","Bhuva Dd","Biochemical And Biophysical Research Communications","Biochimica Et Biophysica Acta - Bioenergetics","Bioconductor Package","Bioinformatics And Biomedical Engineering","Bioinformation","Biology Direct","Bionformatics","BMC Biodata Mining","BMC Cancer","BMC Epigenetics &amp; Chromatin","BMC Infectious Diseases","BMC Medical Genomics","Boileau P","Bolstad Bm","Cancer Informatics","Cancer Research","Cell","Cell Reports Methods","Cell Systems","Chromosome Research","Clinical Epigenetics","Computers In Biology And Medicine","Cuklina J","Current Protocols In Bioinformatics","Dalmolin","Data Integration Life Sci","Database","Developmental Dynamics","Elife","Embnet","Epigenesys - Bioinformatics Protocol","Epigenetics","Epigenetics Chromatin","Fraley C","Front","Frontiers In Molecular Biosciences","Frontiers In Oncology","Genes","Genes &amp; Genomics","Genetics","Genome Biology And Evolution","Genome Medicine","Haibe-Kains B","Hayden N","Henriques D","In Gentleman R","In Lee T","In Mathé E","In Proc","In Schölkopf B","International Journal Of Epidemiology","Iscience","J R Stat Soc Ser C-Appl Stat","Jain N","Japanese J Appl Stat","Journal","Journal Of Alzheimer's Disease","Journal Of Applied Statistics","Journal Of Bioinformatics And Computational Biology","Journal Of Biological Rhythms","Journal Of Cachexia","Journal Of Clinical Bioinformatics","Journal Of Computational And Graphical Statistics","Journal Of Computational Biology","Journal Of Immunological Methods","Journal Of Mass Spectrometry","Journal Of Open Source Software","Journal Of Proteomics","Journal Of Statistical Planning And Infererence","Journal Of the American Statistical Association","Journal Of The International Biometric Society","Journal Of The National Cancer Institute","Journal Of The Royal Statistical Society","Kim V","Kohl M","Kopp W","Lahti L","Lai W","Life Science Alliance","Mallick H","Marini F","Metabolomics","Methods Enzymol","Methods Of Information In Medicine","Molecular Biosystems","Molecular Cell","Nature Biotechnologynol","New Phytologist","Npj Systems Biology And Applications","Oncotarget","Patterns","Paulson Jn","Physiological Genomics","Plants","PLoS Comput","Pollard Ks","Qg14","Raborn Rt","RNA Biology","Robertson Ds","Ruffieux Y","Schlosser P","Source Code Biol Med","Star Protocols","Stark R","Statistica Sinica","Stokowy T","Systems Biomedicine","Tba","Tcbb/Ieee","Technical Report 745","Technometrics","Terfve C","The American Journal Of Human Genetics","The Annals Of Applied Statistics","The Innovation","The Pharmacogenomics Journal","The Prostate","The R Journal","Under Review","University Of Regensburg","Wellcome Open Research","X","Yeung Ky","Zenkova D","Zhang Jd"],[272,90,67,55,31,30,23,20,19,13,11,10,9,8,8,8,7,7,6,6,6,5,5,5,4,4,4,3,3,3,3,3,3,3,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Journal<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><h2 id=session-information>Session information
<a href=#session-information><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#900;font-weight:700>sessionInfo</span>()
</code></pre></div><pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=de_BE.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=de_BE.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=de_BE.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=de_BE.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.8     purrr_0.3.4    
## [5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.6    ggplot2_3.3.5  
## [9] tidyverse_1.3.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.8        lubridate_1.8.0   here_1.0.1        png_0.1-7        
##  [5] rprojroot_2.0.2   assertthat_0.2.1  digest_0.6.29     utf8_1.2.2       
##  [9] R6_2.5.1          cellranger_1.1.0  backports_1.4.1   reprex_2.0.1     
## [13] evaluate_0.15     highr_0.9         httr_1.4.2        blogdown_1.8     
## [17] pillar_1.7.0      rlang_1.0.2       curl_4.3.2        readxl_1.3.1     
## [21] rstudioapi_0.13   jquerylib_0.1.4   DT_0.21           rmarkdown_2.12   
## [25] labeling_0.4.2    selectr_0.4-2     htmlwidgets_1.5.4 munsell_0.5.0    
## [29] broom_0.7.12      compiler_4.1.2    modelr_0.1.8      xfun_0.30        
## [33] pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.2  bookdown_0.24    
## [37] fansi_1.0.2       crayon_1.5.0      tzdb_0.2.0        dbplyr_2.1.1     
## [41] withr_2.5.0       grid_4.1.2        jsonlite_1.8.0    gtable_0.3.0     
## [45] lifecycle_1.0.1   DBI_1.1.2         magrittr_2.0.2    scales_1.1.1     
## [49] cli_3.2.0         stringi_1.7.6     farver_2.1.0      fs_1.5.2         
## [53] xml2_1.3.3        bslib_0.3.1       ellipsis_0.3.2    generics_0.1.2   
## [57] vctrs_0.3.8       tools_4.1.2       glue_1.6.2        crosstalk_1.2.0  
## [61] hms_1.1.1         fastmap_1.1.0     yaml_2.3.5        colorspace_2.0-3 
## [65] rvest_1.0.2       knitr_1.37        haven_2.4.3       patchwork_1.1.1  
## [69] sass_0.4.0
</code></pre><details closed class="f6 fw7 input-reset"><dl class="f6 lh-copy"><dt class=fw7>Posted on:</dt><dd class="fw5 ml0">January 2, 2022</dd></dl><dl class="f6 lh-copy"><dt class=fw7>Length:</dt><dd class="fw5 ml0">8 minute read, 1578 words</dd></dl><dl class="f6 lh-copy"><dt class=fw7>Categories:</dt><dd class="fw5 ml0"><a href=/categories/bioinformatics>bioinformatics</a> <a href=/categories/reproducible-research>reproducible research</a> <a href=/categories/bioconductor>bioconductor</a> <a href=/categories/scientific-writing>scientific writing</a> <a href=/categories/rstats>rstats</a></dd></dl><dl class="f6 lh-copy"><dt class=fw7>See Also:</dt></dl></details></section><footer class=post-footer><div class="post-pagination dt w-100 mt4 mb2"><a class="prev dtc pr2 tl v-top fw6" href=/blog/2021-12-01-push_image_dockerhub/>&larr; Pushing Docker images to Docker Hub</a></div></footer></article></section></main><footer class="site-footer pv4 bt b--transparent ph5" role=contentinfo><nav class="db dt-l w-100"><p class="site-copyright f7 db dtc-l v-mid w-100 w-33-l tc tl-l pv2 pv0-l mv0 lh-copy">&copy; 2022 Fabrício Almeida-Silva
<span class=middot-divider></span>Based on <span xmlns:dct=http://purl.org/dc/terms/ property="dct:title"><a xmlns:dct=http://purl.org/dc/terms/ href=https://github.com/hugo-apero/ rel=dct:source>Hugo Apéro</a></span>.<br>Based on <span xmlns:dct=http://purl.org/dc/terms/ property="dct:title"><a xmlns:dct=http://purl.org/dc/terms/ href=https://github.com/formspree/blogophonic-hugo rel=dct:source>Blogophonic</a></span> by <a xmlns:cc=http://creativecommons.org/ns# href=https://formspree.io property="cc:attributionName" rel=cc:attributionurl>Formspree</a>.</p><div class="site-social-links db dtc-l v-mid w-100 w-33-l tc pv2 pv0-l mv0"><div class=social-icon-links aria-hidden=true><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://github.com/almeidasilvaf title=github target=_blank rel=noopener><i class="fab fa-github fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://twitter.com/almeidasilvaf title=twitter target=_blank rel=noopener><i class="fab fa-twitter fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href="https://scholar.google.com/citations?user=zlnqH0AAAAAJ&hl=en" title=google-scholar target=_blank rel=noopener><i class="ai ai-google-scholar fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=http://lattes.cnpq.br/3119358824056108 title=lattes target=_blank rel=noopener><i class="ai ai-lattes fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://www.researchgate.net/profile/Fabricio-Almeida-Silva title=researchgate target=_blank rel=noopener><i class="ai ai-researchgate fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=/contact/ title=envelope><i class="fas fa-envelope fa-lg fa-fw"></i></a></div></div><div class="site-links f6 db dtc-l v-mid w-100 w-67-l tc tr-l pv2 pv0-l mv0"><a class="dib pv1 ph2 link" href=/license/ title=License>License</a>
<a class="dib pv1 ph2 link" href=/contact/ title="Contact form">Contact</a></div></nav></footer></div></body></html>