<!doctype html><html lang=en dir=ltr><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge,chrome=1"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=generator content="Hugo 0.77.0"><title>Where can I publish a paper describing my Bioconductor package? | Fabrício's personal website</title><meta property="twitter:site" content="@almeidasilvaf"><meta property="twitter:creator" content="@almeidasilvaf"><meta name=description content="Check out where Bioc developers have published their papers"><meta property="og:site_name" content="Fabrício's personal website"><meta property="og:title" content="Where can I publish a paper describing my Bioconductor package? | Fabrício's personal website"><meta property="og:description" content="Check out where Bioc developers have published their papers"><meta property="og:type" content="page"><meta property="og:url" content="/blog/2022-01-03-bioc_publications/"><meta property="og:locale" content="en"><meta property="og:image" content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta property="twitter:card" content="summary_large_image"><meta name=twitter:image content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta itemprop=name content="Where can I publish a paper describing my Bioconductor package?"><meta itemprop=description content="Motivation     When I developed BioNERO, my first R/Bioconductor package, I didn’t know to which journals I could submit the paper describing it. Since then, I’ve seen many other R developers that have faced the same issue. To help solve this problem, here I will teach you how to do some web scraping to find out the main journals where Bioc developers publish papers describing their packages."><meta itemprop=datePublished content="2022-01-02T00:00:00+00:00"><meta itemprop=dateModified content="2022-01-02T00:00:00+00:00"><meta itemprop=wordCount content="1724"><meta itemprop=image content="/blog/2022-01-03-bioc_publications/featured-bioc.png"><meta itemprop=keywords content><script type=application/javascript>var dnt=(navigator.doNotTrack||window.doNotTrack||navigator.msDoNotTrack);var doNotTrack=(dnt=="1"||dnt=="yes");if(!doNotTrack){window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;ga('create','G-4M49R7N8QJ','auto');ga('set','anonymizeIp',true);ga('send','pageview');}</script><script async src=https://www.google-analytics.com/analytics.js></script><!--[if IE]><script src=//html5shiv.googlecode.com/svn/trunk/html5.js></script><![endif]--><link rel="shortcut icon" href=/img/logo_tree_graph.png type=image/x-icon><link rel=icon href=/img/logo_tree_graph.png type=image/x-icon><link rel=stylesheet href=/style.main.min.306f61a3dc4566d041ed43ec1724f89fc7dfd8c53cd1b942e7b37a3b790db35b.css integrity="sha256-MG9ho9xFZtBB7UPsFyT4n8ff2MU80blC57N6O3kNs1s=" media=screen><script src=/panelset.min.41f62c42d5f25dd80e81136866f4f1baa1bc17669edfd002e75c0e107953789b.js type=text/javascript></script></head><body><div class="grid-container single"><header class="site-header pt4 pb2 mb4 bb b--transparent ph5 headroom z-max" role=banner><script>MathJax={tex:{inlineMath:[['$','$'],['\\(','\\)']],displayMath:[['$$','$$'],['\\[','\\]']],processEscapes:true,processEnvironments:true},options:{skipHtmlTags:['script','noscript','style','textarea','pre']}};window.addEventListener('load',(event)=>{document.querySelectorAll("mjx-container").forEach(function(x){x.parentElement.classList+='has-jax'})});</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script type=text/javascript id=MathJax-script async src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script><nav class="site-nav db dt-l w-100" role=navigation><a class="site-brand db dtc-l v-mid link no-underline w-100 w-33-l tc tl-l" href=/ title=Home><img src=/img/logo_tree_graph.png class="dib db-l h2 w-auto" alt="Fabrício's personal website"></a><div class="site-links db dtc-l v-mid w-100 w-47-l tc tr-l mt3 mt0-l ttu tracked"><a class="link f6 f5-l dib pv1 ph2" href=/about/ title="About me">About</a>
<a class="link f6 f5-l dib pv1 ph2" href=/project/ title=Softwares>Softwares & Data</a>
<a class="link f6 f5-l dib pv1 ph2" href=/publications/ title=Publications>Publications</a>
<a class="link f6 f5-l dib pv1 ph2" href=/talk/ title=Talks>Talks</a>
<a class="link f6 f5-l dib pv1 ph2" href=/teaching/ title=Teaching>Teaching</a>
<a class="link f6 f5-l dib pv1 ph2" href=/cv/ title=CV>CV</a>
<a class="link f6 f5-l dib pv1 ph2 active" href=/blog/ title=Blog>Blog</a></div></nav></header><main class="page-main pa4" role=main><section class="page-content mw7 center"><article class="post-content pa0 ph4-l"><header class=post-header><h1 class="f1 lh-solid measure-narrow mb3 fw4">Where can I publish a paper describing my Bioconductor package?</h1><p class="f6 measure lh-copy mv1">By Fabrício Almeida-Silva in <a href=/categories/bioinformatics>bioinformatics</a> <a href=/categories/reproducible-research>reproducible research</a> <a href=/categories/bioconductor>bioconductor</a> <a href=/categories/scientific-writing>scientific writing</a> <a href=/categories/rstats>rstats</a></p><p class="f7 db mv0 ttu">January 2, 2022</p></header><section class="post-body pt5 pb4"><script src=/blog/2022-01-03-bioc_publications/index_files/htmlwidgets/htmlwidgets.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/datatables-css/datatables-crosstalk.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/datatables-binding/datatables.js></script><script src=/blog/2022-01-03-bioc_publications/index_files/jquery/jquery-3.6.0.min.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/dt-core/css/jquery.dataTables.min.css rel=stylesheet><link href=/blog/2022-01-03-bioc_publications/index_files/dt-core/css/jquery.dataTables.extra.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/dt-core/js/jquery.dataTables.min.js></script><link href=/blog/2022-01-03-bioc_publications/index_files/crosstalk/css/crosstalk.css rel=stylesheet><script src=/blog/2022-01-03-bioc_publications/index_files/crosstalk/js/crosstalk.min.js></script><h2 id=motivation>Motivation
<a href=#motivation><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>When I developed
<a href=https://bioconductor.org/packages/release/bioc/html/BioNERO.html target=_blank rel=noopener>BioNERO</a>, my first R/Bioconductor package, I didn’t know to which journals I could submit the paper describing it. Since then, I’ve seen many other R developers that have faced the same issue. To help solve this problem, here I will teach you how to do some web scraping to find out the main journals where Bioc developers publish papers describing their packages.</p><h2 id=extracting-citation-information-from-packages-landing-pages>Extracting citation information from packages’ landing pages
<a href=#extracting-citation-information-from-packages-landing-pages><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>All Bioconductor packages have a landing page where there is a citation field. This field is created from a CITATION file inside the package directory or automatically from information in the DESCRIPTION file. First, let’s write two functions to helper functions:</p><ul><li><code>get_bioc_citation</code>, a function that scrapes packages’ landing pages to get citation information.</li><li><code>create_citation_table</code>, a function that created a tidy data frame containing package name, journal where the paper is published (if that is the case), and year of publication.</li></ul><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#998;font-style:italic>#&#39; Extract citation from Bioconductor landing page</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @param package Character of package name.</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @return Character of citation text.</span>
<span style=color:#998;font-style:italic>#&#39; @noRd</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom rvest html_elements html_text2</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom httr GET content</span>
get_bioc_citation <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>function</span>(package <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>NULL</span>) {
    url <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>paste0</span>(<span style=color:#d14>&#34;https://bioconductor.org/packages/release/bioc/citations/&#34;</span>,
                  package, <span style=color:#d14>&#34;/citation.html&#34;</span>)
    request <span style=color:#000;font-weight:700>&lt;-</span> httr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>GET</span>(url)
    parsed_page <span style=color:#000;font-weight:700>&lt;-</span> httr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>content</span>(request, as <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;parsed&#34;</span>, encoding <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;UTF-8&#34;</span>)
    <span style=color:#900;font-weight:700>if</span>(<span style=color:#d14>&#34;xml_node&#34;</span> <span style=color:#000;font-weight:700>%in%</span> <span style=color:#900;font-weight:700>class</span>(parsed_page)) {
        citations <span style=color:#000;font-weight:700>&lt;-</span> rvest<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>html_elements</span>(parsed_page, <span style=color:#d14>&#34;p&#34;</span>)
        citations <span style=color:#000;font-weight:700>&lt;-</span> rvest<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>html_text2</span>(citations)
    } else {
        citations <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#000;font-weight:700>NA</span>
    }
    <span style=color:#900;font-weight:700>return</span>(citations)
}

<span style=color:#998;font-style:italic>#&#39; Create a table of citations for each Bioc package</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @param packages Character vector of package names.</span>
<span style=color:#998;font-style:italic>#&#39;</span>
<span style=color:#998;font-style:italic>#&#39; @return A data frame with the following variables:</span>
<span style=color:#998;font-style:italic>#&#39; \describe{</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Package}{Chaarcter of package name.}</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Journal}{Character of journal name.}</span>
<span style=color:#998;font-style:italic>#&#39;   \item{Year}{Numeric of journal year.}</span>
<span style=color:#998;font-style:italic>#&#39; }</span>
<span style=color:#998;font-style:italic>#&#39; @noRd</span>
<span style=color:#998;font-style:italic>#&#39; @importFrom stringr str_detect str_extract</span>
create_citation_table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>function</span>(packages <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>NULL</span>) {
    table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>Reduce</span>(rbind, <span style=color:#900;font-weight:700>lapply</span>(packages, <span style=color:#900;font-weight:700>function</span>(x) {
        df <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>data.frame</span>(Citation <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>get_bioc_citation</span>(x))
        df<span style=color:#000;font-weight:700>$</span>Package <span style=color:#000;font-weight:700>&lt;-</span> x
        <span style=color:#900;font-weight:700>return</span>(df)
    }))
    
    year <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;([0-9]{4})&#34;</span>
    table<span style=color:#000;font-weight:700>$</span>Year <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>as.numeric</span>(stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_extract</span>(table<span style=color:#000;font-weight:700>$</span>Citation, year))
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#39;.*” &#39;</span>, <span style=color:#d14>&#39;&#39;</span>, table<span style=color:#000;font-weight:700>$</span>Citation)
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#34;\\*{3} &#34;</span>, <span style=color:#d14>&#34;&#34;</span>, journal)
    journal <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>gsub</span>(<span style=color:#d14>&#34;(,|\\.).*&#34;</span>, <span style=color:#d14>&#34;&#34;</span>, journal)
    table<span style=color:#000;font-weight:700>$</span>Journal <span style=color:#000;font-weight:700>&lt;-</span> journal
    idx <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>which</span>(stringr<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>str_detect</span>(table<span style=color:#000;font-weight:700>$</span>Citation, <span style=color:#d14>&#34;github|R package version&#34;</span>))
    table<span style=color:#000;font-weight:700>$</span>Journal[idx] <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#000;font-weight:700>NA</span>
    table <span style=color:#000;font-weight:700>&lt;-</span> table[, <span style=color:#900;font-weight:700>c</span>(<span style=color:#d14>&#34;Package&#34;</span>, <span style=color:#d14>&#34;Journal&#34;</span>, <span style=color:#d14>&#34;Year&#34;</span>)]
    table <span style=color:#000;font-weight:700>&lt;-</span> table[<span style=color:#000;font-weight:700>!</span><span style=color:#900;font-weight:700>is.na</span>(table<span style=color:#000;font-weight:700>$</span>Journal), ]
    <span style=color:#900;font-weight:700>return</span>(table)
}
</code></pre></div><p>Now, we can create the citation table.</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#900;font-weight:700>suppressPackageStartupMessages</span>(<span style=color:#900;font-weight:700>library</span>(tidyverse))

<span style=color:#998;font-style:italic># List available Bioc packages</span>
bioc_repo <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#d14>&#34;https://www.bioconductor.org/packages/release/bioc&#34;</span>
bioc_pkgs <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>available.packages</span>(repos <span style=color:#000;font-weight:700>=</span> bioc_repo)[, <span style=color:#d14>&#34;Package&#34;</span>]

<span style=color:#998;font-style:italic># Scrape CITATION field from Bioc landing pages</span>
citation_table <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>create_citation_table</span>(bioc_pkgs)
</code></pre></div><h2 id=summary-stats>Summary stats
<a href=#summary-stats><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><p>Now, let’s count the frequency of packages in each journal and show the top 20 journals based on highest article count.</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>citation_table <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>count</span>(Journal) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>arrange</span>(<span style=color:#000;font-weight:700>-</span>n) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>slice_head</span>(n <span style=color:#000;font-weight:700>=</span> <span style=color:#099>20</span>)
</code></pre></div><pre><code>##                       Journal   n
## 1              Bioinformatics 262
## 2          BMC Bioinformatics  82
## 3                     bioRxiv  63
## 4              Genome Biology  51
## 5      Nucleic Acids Research  42
## 6              Nature Methods  29
## 7               F1000Research  25
## 8                BMC Genomics  22
## 9                    PLoS ONE  13
## 10            Genome Research  12
## 11          Nucleic Acids Res  12
## 12              Biostatistics  10
## 13         BMC bioinformatics  10
## 14       Analytical Chemistry   9
## 15                        Nat   7
## 16       Nature Biotechnology   7
## 17      Nature Communications   7
## 18 PLoS Computational Biology   7
## 19     Bioinformatics (Oxford   6
## 20                    doi: 10   6
</code></pre><p>We can see that there are some issues, such as different names for the same journal, the preprint server bioRxiv listed as a journal, and a weird ‘doi’ journal that is likely due to manual insertion of citation info. Let’s fix them.</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>citation_stats <span style=color:#000;font-weight:700>&lt;-</span> citation_table <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>mutate</span>(Journal <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>str_replace_all</span>(
        Journal, <span style=color:#900;font-weight:700>c</span>(<span style=color:#d14>&#34;Nat$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Nature&#34;</span>,
                   <span style=color:#d14>&#34;Res$&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Research&#34;</span>,
                   <span style=color:#d14>&#34; \\(Oxford.*&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>,
                   <span style=color:#d14>&#34;bioinformatics&#34;</span> <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Bioinformatics&#34;</span>))
    ) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>count</span>(Journal) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>arrange</span>(<span style=color:#000;font-weight:700>-</span>n) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>filter</span>(<span style=color:#000;font-weight:700>!</span>Journal <span style=color:#000;font-weight:700>%in%</span> <span style=color:#900;font-weight:700>c</span>(<span style=color:#d14>&#34;bioRxiv&#34;</span>, <span style=color:#d14>&#34;BioRxiv&#34;</span>, <span style=color:#d14>&#34;doi: 10&#34;</span>)) <span style=color:#000;font-weight:700>%&gt;%</span>
    <span style=color:#900;font-weight:700>slice_head</span>(n <span style=color:#000;font-weight:700>=</span> <span style=color:#099>20</span>)
citation_stats
</code></pre></div><pre><code>##                            Journal   n
## 1                   Bioinformatics 268
## 2               BMC Bioinformatics  92
## 3           Nucleic Acids Research  54
## 4                   Genome Biology  51
## 5                   Nature Methods  29
## 6                    F1000Research  27
## 7                     BMC Genomics  22
## 8                  Genome Research  13
## 9                         PLoS ONE  13
## 10                   Biostatistics  10
## 11            Analytical Chemistry   9
## 12                          Nature   8
## 13            Nature Biotechnology   7
## 14           Nature Communications   7
## 15      PLoS Computational Biology   7
## 16    Journal of Proteome Research   6
## 17       Molecular Systems Biology   6
## 18 Journal of Statistical Software   5
## 19 Molecular &amp; Cellular Proteomics   5
## 20                 Nature Genetics   5
</code></pre><p>Exploring it visually:</p><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r>bioc_logo <span style=color:#000;font-weight:700>&lt;-</span> png<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>readPNG</span>(
    here<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>here</span>(<span style=color:#d14>&#34;content&#34;</span>, <span style=color:#d14>&#34;blog&#34;</span>, 
               <span style=color:#d14>&#34;2022-01-03-bioc_publications&#34;</span>, 
               <span style=color:#d14>&#34;featured-bioc.png&#34;</span>), 
    native <span style=color:#000;font-weight:700>=</span> <span style=color:#000;font-weight:700>TRUE</span>)

last_updated <span style=color:#000;font-weight:700>&lt;-</span> <span style=color:#900;font-weight:700>format</span>(<span style=color:#900;font-weight:700>Sys.Date</span>(), <span style=color:#d14>&#34;%Y-%m-%d&#34;</span>)

<span style=color:#900;font-weight:700>ggplot</span>(citation_stats, <span style=color:#900;font-weight:700>aes</span>(x <span style=color:#000;font-weight:700>=</span> n, y <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>reorder</span>(Journal, n))) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>geom_col</span>() <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>geom_text</span>(<span style=color:#900;font-weight:700>aes</span>(label <span style=color:#000;font-weight:700>=</span> n), hjust <span style=color:#000;font-weight:700>=</span> <span style=color:#099>-0.3</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>xlim</span>(<span style=color:#099>0</span>, <span style=color:#099>300</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>labs</span>(title <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Where are papers associated with BioC packages published?&#34;</span>,
         subtitle <span style=color:#000;font-weight:700>=</span> <span style=color:#900;font-weight:700>paste0</span>(<span style=color:#d14>&#34;Last updated: &#34;</span>, last_updated),
         x <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;Number of papers&#34;</span>, y <span style=color:#000;font-weight:700>=</span> <span style=color:#d14>&#34;&#34;</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>theme_bw</span>() <span style=color:#000;font-weight:700>+</span>
    patchwork<span style=color:#000;font-weight:700>::</span><span style=color:#900;font-weight:700>inset_element</span>(bioc_logo,
                             left <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.5</span>,
                             bottom <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.55</span>,
                             right <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.95</span>,
                             top <span style=color:#000;font-weight:700>=</span> <span style=color:#099>0.3</span>) <span style=color:#000;font-weight:700>+</span>
    <span style=color:#900;font-weight:700>theme_void</span>()
</code></pre></div><p><img src=/blog/2022-01-03-bioc_publications/index_files/figure-html/unnamed-chunk-4-1.png width=720></p><p>And voilà! In case you want to explore the whole table, here it is:</p><div id=htmlwidget-1 style=width:100%;height:auto class="datatables html-widget"></div><script type=application/json data-for=htmlwidget-1>{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271"],["Bioinformatics","BMC Bioinformatics","Nucleic Acids Research","Genome Biology","Nature Methods","F1000Research","BMC Genomics","Genome Research","PLoS ONE","Biostatistics","Analytical Chemistry","Nature","Nature Biotechnology","Nature Communications","PLoS Computational Biology","Journal of Proteome Research","Molecular Systems Biology","Journal of Statistical Software","Molecular &amp; Cellular Proteomics","Nature Genetics","Nucl","PLoS computational biology","BMC Systems Biology","Briefings in Bioinformatics","Frontiers in Genetics","Nucleic acids research","PLOS Computational Biology","Proceedings of the National Academy of Sciences","Statistical Applications in Genetics and Molecular Biology","Annals of Applied Statistics","Biometrics","Cell Reports","Cytometry Part A","Genome Biol","Johns Hopkins University","Molecular Biology and Evolution","PLOS ONE","Aibar S","Analytica Chimica Acta","Animal Genetics","BIOINFORMATICS","BioMed Research International","Clinical Cancer Research","Epigenetics &amp; Chromatin","Epigenomics","F1000","Frontiers in genetics","Genomics","Gigascience","Human Mutation","in preparation","Int J Mol Sci","J","Journal of Biomedical Informatics","Luo X","manuscript in preparation","Manuscript in preparation","Methods","Methods in Ecology and Evolution","Methods in Molecular Biology","Microbiome","Mol","Mol Cell Proteomics","Nature communications","Nature Protocols","Nucleic Acids Research (Database issue)","OMICS: A Journal of Integrative Biology","PeerJ","PLoS Comput Biol","PLoS One","PNAS","Scientific reports","Scientific Reports","Shokoohi F (2020)","Source Code for Biology and Medicine","Stat Appl Genet Mol Biol","The Journal of Open Source Software","","10","Aging (Albany NY)","AISTATS","Algorithmica","Algorithms for Molecular Biology","Algorithms Mol Biol","American Journal of Human Genetics","Ann","Applications in Genetics and Molecular Biology","Applications in Plant Sciences","arxiv","arXiv","arXiv preprint arXiv:1805","Baek M","BBAGRM","Bhuva DD","Biochemical and Biophysical Research Communications","Biochimica et Biophysica Acta (BBA) - Bioenergetics","Bioconductor Package","Bioinformatics and Biomedical Engineering","Bioinformatics Journal","Bioinformation","Biology Direct","Bionformatics","biorxiv","biorXiv","Biostatistics (accepted)","BMC BioData Mining","BMC Cancer","BMC Epigenetics &amp; Chromatin","BMC genomics","BMC Infectious Diseases","BMC Medical Genomics","BMC Syst Biol","Boileau P","Bolstad BM (2004)","Cancer Informatics","Cancer Research","Cell","Cell Reports Methods","Cell Systems","Chromosome Research","Clinical Epigenetics","Comput Methods Programs Biomed","Computer Methods and Programs in Biomedicine","Computers in Biology and Medicine","Cuklina J","Current Protocols in Bioinformatics","Dalmolin","Data Integration Life Sci","Database","Developmental Dynamics","eLife","EMBnet","Epigenesys - Bioinformatics Protocol","Epigenetics","Epigenetics Chromatin","F1000 Research","f1000research","Fraley C","Front","Frontiers in Molecular Biosciences","Frontiers in Oncology","Genes","Genes &amp; Genomics","Genetics","Genome Biology and Evolution","Genome Medicine","Haibe-Kains B","Hayden N","Henriques D","http://plato","http://www","https://doi","Important note to the maintainer of the BRAIN package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the ChemmineR package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the clippda package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the diggit package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the DMRcate package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the eiR package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the flowFP package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the fmcsR package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the GOSim package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the GSRI package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the MEDME package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the pandaR package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the rfPred package: An error occured while trying to generate the citation from the CITATION file","Important note to the maintainer of the SIM package: An error occured while trying to generate the citation from the CITATION file","In Gentleman R","In Lee T","In Mathé E","In Proc","In Schölkopf B","International journal of epidemiology","iScience","J Comput Biol","J Mach Learn Research","J Proteomics","J R Stat Soc Ser C-Appl Stat","Jain N","Japanese J Appl Stat","JASA","JCO Clinical Cancer Informatics","JCO Precision Oncology","Journal","Journal of Alzheimer's Disease","Journal of Applied Statistics","Journal of Bioinformatics and Computational Biology","Journal of Biological Rhythms","Journal of cachexia","Journal of Clinical Bioinformatics","Journal of Computational and Graphical Statistics","Journal of Immunological Methods","Journal of Machine Learning Research","Journal of Mass Spectrometry","Journal of Open Source Software","Journal of Statistical Planning and Infererence","Journal of the International Biometric Society","Journal of the National Cancer Institute","Journal of the Royal Statistical Society","Kim V (2021)","Kohl M (2007)","Kopp W (2017)","Lahti L","Lai W","Life Science Alliance","Mallick H","Manuscript submitted for publication","Marini F","Metabolomics","Methods Enzymol","Methods Inf Med","Mol Syst Biol","Molecular BioSystems","Molecular Cell","Molecular systems biology","NAR Genomics and Bioinformatics","Nat Biotech","Nat Biotechnol","Nat Commun","Nat Meth","Nature genetics","New Phytologist","npj Systems Biology and Applications","Oncotarget","Oxford Bioinformatics","Patterns","Paulson JN","PeerJ Preprints","Physiol Genomics","Plants","PLoS Comput","PLoS Computat Biol","PLoS Genetics","Plos one","PLoS one","Pollard KS","Preprint available in https://www","Preprint in Mathematical Sciences 18","Proc Natl Acad Sci USA","Proceedings of the National Academy of Sciences of the United States of America","QG14","Raborn RT","RNA Biology","Robertson DS","Ruffieux Y","SAGMB","Schlosser P","Source Code Biol Med","STAR Protocols","Stark R","Statistica Sinica","Stokowy T (2016)","Submitted","Systems Biomedicine","tba","TCBB/IEEE","Technical Report 745","Technometrics","Terfve C","The American Journal of Human Genetics","The Annals of Applied Statistics","The Innovation","The Pharmacogenomics Journal","The Prostate","The R Journal","Under review","University of Regensburg","Wellcome open research","X","Yeung KY","Zenkova D","Zhang JD (2021)"],[268,92,54,51,29,27,22,13,13,10,9,8,7,7,7,6,6,5,5,5,5,5,4,4,4,4,4,4,4,3,3,3,3,3,3,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Journal<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><h2 id=session-information>Session information
<a href=#session-information><svg class="anchor-symbol" aria-hidden="true" height="26" width="26" viewBox="0 0 22 22" xmlns="http://www.w3.org/2000/svg"><path d="M0 0h24v24H0z" fill="currentcolor"/><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76.0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71.0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71.0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76.0 5-2.24 5-5s-2.24-5-5-5z"/></svg></a></h2><div class=highlight><pre style=background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=color:#900;font-weight:700>sessionInfo</span>()
</code></pre></div><pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4    
## [5] readr_2.0.2     tidyr_1.1.4     tibble_3.1.5    ggplot2_3.3.5  
## [9] tidyverse_1.3.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.7        here_1.0.1        lubridate_1.8.0   png_0.1-7        
##  [5] rprojroot_2.0.2   assertthat_0.2.1  digest_0.6.29     utf8_1.2.2       
##  [9] R6_2.5.1          cellranger_1.1.0  backports_1.2.1   reprex_2.0.1     
## [13] evaluate_0.14     highr_0.9         httr_1.4.2        blogdown_1.5     
## [17] pillar_1.6.4      rlang_0.4.12      curl_4.3.2        readxl_1.3.1     
## [21] rstudioapi_0.13   jquerylib_0.1.4   DT_0.19           rmarkdown_2.11   
## [25] labeling_0.4.2    selectr_0.4-2     htmlwidgets_1.5.4 munsell_0.5.0    
## [29] broom_0.7.9       compiler_4.1.0    modelr_0.1.8      xfun_0.29        
## [33] pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.1  bookdown_0.24    
## [37] codetools_0.2-18  fansi_0.5.0       crayon_1.4.1      tzdb_0.1.2       
## [41] dbplyr_2.1.1      withr_2.4.2       grid_4.1.0        jsonlite_1.7.2   
## [45] gtable_0.3.0      lifecycle_1.0.1   DBI_1.1.1         magrittr_2.0.1   
## [49] scales_1.1.1      cli_3.0.1         stringi_1.7.6     farver_2.1.0     
## [53] fs_1.5.2          xml2_1.3.2        bslib_0.3.1       ellipsis_0.3.2   
## [57] generics_0.1.0    vctrs_0.3.8       tools_4.1.0       glue_1.6.0       
## [61] crosstalk_1.1.1   hms_1.1.1         fastmap_1.1.0     yaml_2.2.1       
## [65] colorspace_2.0-2  rvest_1.0.2       knitr_1.37        haven_2.4.3      
## [69] patchwork_1.1.1   sass_0.4.0
</code></pre><details closed class="f6 fw7 input-reset"><dl class="f6 lh-copy"><dt class=fw7>Posted on:</dt><dd class="fw5 ml0">January 2, 2022</dd></dl><dl class="f6 lh-copy"><dt class=fw7>Length:</dt><dd class="fw5 ml0">9 minute read, 1724 words</dd></dl><dl class="f6 lh-copy"><dt class=fw7>Categories:</dt><dd class="fw5 ml0"><a href=/categories/bioinformatics>bioinformatics</a> <a href=/categories/reproducible-research>reproducible research</a> <a href=/categories/bioconductor>bioconductor</a> <a href=/categories/scientific-writing>scientific writing</a> <a href=/categories/rstats>rstats</a></dd></dl><dl class="f6 lh-copy"><dt class=fw7>See Also:</dt></dl></details></section><footer class=post-footer><div class="post-pagination dt w-100 mt4 mb2"><a class="prev dtc pr2 tl v-top fw6" href=/blog/2021-12-01-push_image_dockerhub/>&larr; Pushing Docker images to Docker Hub</a></div></footer></article></section></main><footer class="site-footer pv4 bt b--transparent ph5" role=contentinfo><nav class="db dt-l w-100"><p class="site-copyright f7 db dtc-l v-mid w-100 w-33-l tc tl-l pv2 pv0-l mv0 lh-copy">&copy; 2022 Fabrício Almeida-Silva
<span class=middot-divider></span>Based on <span xmlns:dct=http://purl.org/dc/terms/ property="dct:title"><a xmlns:dct=http://purl.org/dc/terms/ href=https://github.com/hugo-apero/ rel=dct:source>Hugo Apéro</a></span>.<br>Based on <span xmlns:dct=http://purl.org/dc/terms/ property="dct:title"><a xmlns:dct=http://purl.org/dc/terms/ href=https://github.com/formspree/blogophonic-hugo rel=dct:source>Blogophonic</a></span> by <a xmlns:cc=http://creativecommons.org/ns# href=https://formspree.io property="cc:attributionName" rel=cc:attributionurl>Formspree</a>.</p><div class="site-social-links db dtc-l v-mid w-100 w-33-l tc pv2 pv0-l mv0"><div class=social-icon-links aria-hidden=true><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://github.com/almeidasilvaf title=github target=_blank rel=noopener><i class="fab fa-github fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://twitter.com/almeidasilvaf title=twitter target=_blank rel=noopener><i class="fab fa-twitter fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href="https://scholar.google.com/citations?user=zlnqH0AAAAAJ&hl=en" title=google-scholar target=_blank rel=noopener><i class="ai ai-google-scholar fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=http://lattes.cnpq.br/3119358824056108 title=lattes target=_blank rel=noopener><i class="ai ai-lattes fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=https://www.researchgate.net/profile/Fabricio-Almeida-Silva title=researchgate target=_blank rel=noopener><i class="ai ai-researchgate fa-lg fa-fw"></i></a><a class="link dib h1 w1 ml0 mr2 f6 o-90 glow" href=/contact/ title=envelope><i class="fas fa-envelope fa-lg fa-fw"></i></a></div></div><div class="site-links f6 db dtc-l v-mid w-100 w-67-l tc tr-l pv2 pv0-l mv0"><a class="dib pv1 ph2 link" href=/license/ title=License>License</a>
<a class="dib pv1 ph2 link" href=/contact/ title="Contact form">Contact</a></div></nav></footer></div></body></html>